lib/sarif/bandit_sarif.rb

```diff
@@ -0,0 +1,57 @@
+module Sarif
+  class BanditSarif < BaseSarif
+    BANDIT_URI = 'https://github.com/PyCQA/bandit'.freeze
+
+    def initialize(scan_report)
+      super(scan_report)
+      @uri = BANDIT_URI
+      begin
+        @logs = JSON.parse(@scan_report.log(''))['results']
+      rescue JSON::ParserError
+        @logs = []
+      end
+      @unique_issues = Set.new
+    end
+
+    def parse_issue(issue)
+      key = ""
+      key << issue["filename"].upcase + issue["line_number"].to_s + issue['issue_text']
+      return nil if @unique_issues.include? key
+
+      @unique_issues.add(key)
+      endline = issue['line_range'][issue['line_range'].size - 1]
+      {
+        id: issue['test_id'],
+        name: issue['test_name'],
+        level: issue['issue_severity'],
+        details: "#{issue['issue_text']} \nissue_confidence: #{issue['issue_severity']}"\
+        "\nissue_severity: #{issue['issue_severity']}",
+        start_line: issue["line_number"].to_i,
+        end_line: endline,
+        start_column: 1,
+        uri: issue["filename"],
+        help_url: issue['more_info'],
+        code: issue['code']
+      }
+    end
+
+    def build_invocations
+      if @logs.empty? && !@scan_report.passed?
```

Comment:
Why do you check `if @log.empty?`
Is it not sufficient to only check `if !@scan_report.passed?`

Reply:
if the bandit scanner fails, it could mean two things: 
1) vulnerabilities were found
2) errors were reported by bandit while scanning 

to catch the generated error case, the logs generated by the scanner would be empty and the scanner would fail. This behaviour means that an error was generated. 

However, if the logs were empty and the scanner passed, then no errors are generated

Reply:
Sounds good

Reply:
Errors in the json are in the format 
```
{
  error: 'error name' 
  reason: 'reason for erro' 
} 
```
they are now logged in the sarif report